src/modules/git_metrics.rs

```diff
@@ -0,0 +1,288 @@
+use regex::Regex;
+
+use crate::{
+    config::RootModuleConfig, configs::git_metrics::GitMetricsConfig, formatter::StringFormatter,
+    module::Module,
+};
+
+use super::Context;
+
+/// Creates a module with the current added/modified/deleted lines in the git repository at the
+/// current directory
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("git_metrics");
+    let config: GitMetricsConfig = GitMetricsConfig::try_load(module.config);
+
+    let repo = context.get_repo().ok()?;
+    let repo_root = repo.root.as_ref()?;
+
+    let diff = context
+        .exec_cmd(
+            "git",
+            &[
+                "-C",
+                &repo_root.to_string_lossy(),
+                "--no-optional-locks",
+                "diff",
+                "--word-diff",
+                "--unified=0",
+            ],
+        )?
+        .stdout;
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_style(|variable| match variable {
+                "a_style" => Some(Ok(config.a_style)),
+                "m_style" => Some(Ok(config.m_style)),
+                "d_style" => Some(Ok(config.d_style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "added" => Some(Ok(get_added_lines(&diff)?)),
+                "modified" => Some(Ok(get_modified_lines(&diff)?)),
+                "deleted" => Some(Ok(get_deleted_lines(&diff)?)),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `git_metrics`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_added_lines(diff: &str) -> Option<String> {
+    let re = Regex::new(r"^\{\+.*\+\}$").unwrap();
+
+    Some(
+        diff.lines()
+            .filter(|line| re.is_match(line))
+            .count()
+            .to_string(),
+    )
+}
+fn get_modified_lines(diff: &str) -> Option<String> {
+    let re = Regex::new(r"^.+(\[-|\{\+).*$").unwrap();
+
+    Some(
+        diff.lines()
+            .filter(|line| re.is_match(line))
+            .count()
+            .to_string(),
+    )
+}
+fn get_deleted_lines(diff: &str) -> Option<String> {
+    let re = Regex::new(r"^\[-.*-\]$").unwrap();
+
+    Some(
+        diff.lines()
+            .filter(|line| re.is_match(line))
+            .count()
+            .to_string(),
+    )
+}
```

Comment:
Instead of iterating through the diff three times, it may be simpler and more declarative to have a `GitDiff` struct that parses output a single time:

```rust
struct GitDiff {
  added: u32,
  modified: u32,
  deleted: u32,
}

impl GitDiff {
  fn parse(diff: &str) -> Self;
}

```